vispy/visuals/volume.py

```diff
@@ -393,29 +395,29 @@
 
 _ATTENUATED_MIP_SNIPPETS = dict(
     before_loop="""
-        float maxval = -99999.0; // The maximum encountered value
+        float maxval = u_mip_cutoff; // The maximum encountered value
         float sumval = 0.0; // The sum of the encountered values
         float scaled = 0.0; // The scaled value
-        int maxi = 0;  // Where the maximum value was encountered
-        vec3 maxloc = vec3(0.0);  // Location where the maximum value was encountered
+        int maxi = -1;  // Where the maximum value was encountered
         """,
     in_loop="""
         sumval = sumval + val;
         scaled = val * exp(-u_attenuation * (sumval - 1) / u_relative_step_size);
         if( scaled > maxval ) {
             maxval = scaled;
             maxi = iter;
-            maxloc = loc;
         }
         """,
     after_loop="""
-        gl_FragColor = applyColormap(maxval);
+        if (maxi > -1) {
+            gl_FragColor = applyColormap(maxval);
+        }
```

Comment:
Keeping in mind that I don't know much about volume rendering, is this just an optimization or does it actually fix things showing up when they shouldn't be there?